src/hotspot/cpu/ppc/globals_ppc.hpp

```diff
@@ -81,73 +81,74 @@ define_pd_global(intx, InitArrayShortSize, 9*BytesPerLong);
-  product(bool, SuperwordUseVSX, false,                                     \
-          "Use Power8 VSX instructions for superword optimization.")        \
-                                                                            \
   /* Reoptimize code-sequences of calls at runtime, e.g. replace an */      \
   /* indirect call by a direct call.                                */      \
-  product(bool, ReoptimizeCallSequences, true,                              \
+  product(bool, ReoptimizeCallSequences, true, DIAGNOSTIC,                  \
           "Reoptimize code-sequences of calls at runtime.")                 \
                                                                             \
   /* Power 8: Configure Data Stream Control Register. */                    \
-  product(uint64_t,DSCR_PPC64, (uintx)-1,                                   \
+  product(uint64_t, DSCR_PPC64, (uint64_t)-1,                               \
           "Power8 or later: Specify encoded value for Data Stream Control " \
           "Register")                                                       \
-  product(uint64_t,DSCR_DPFD_PPC64, 8,                                      \
+  product(uint64_t, DSCR_DPFD_PPC64, 8,                                     \
           "Power8 or later: DPFD (default prefetch depth) value of the "    \
           "Data Stream Control Register."                                   \
           " 0: hardware default, 1: none, 2-7: min-max, 8: don't touch")    \
-  product(uint64_t,DSCR_URG_PPC64, 8,                                       \
+  product(uint64_t, DSCR_URG_PPC64, 8,                                      \
           "Power8 or later: URG (depth attainment urgency) value of the "   \
           "Data Stream Control Register."                                   \
           " 0: hardware default, 1: none, 2-7: min-max, 8: don't touch")    \
                                                                             \
-  product(bool, UseLoadInstructionsForStackBangingPPC64, false,             \
+  product(bool, UseLoadInstructionsForStackBangingPPC64, false, DIAGNOSTIC, \
           "Use load instructions for stack banging.")                       \
                                                                             \
+  product(bool, UseStaticBranchPredictionInCompareAndSwapPPC64, true, DIAGNOSTIC,\
+          "Use static branch prediction hints in CAS operations.")          \
+  product(bool, UseStaticBranchPredictionForUncommonPathsPPC64, false, DIAGNOSTIC,\
+          "Use static branch prediction hints for uncommon paths.")         \
+                                                                            \
   /* special instructions */                                                \
-  product(bool, UseByteReverseInstructions, false,                          \
+  product(bool, SuperwordUseVSX, false,                                     \
```

Comment:
Why leave this one out?

Reply:
SuperwordUseVSX switches usage of vector registers in C2. It's more fundamental than simple instruction usage switches. It's comparable to UseAVX on x86 which is product, too. So I prefer keeping it product for now. Are you ok with it?

Reply:
Sure, just curious. Leave it up to you.